components/search/SearchCollection.vue

```diff
@@ -45,162 +45,142 @@
 })
-export default class SearchCollection extends mixins(KeyboardEventsMixin) {
-  @Prop(String) public search!: string
-  @Prop(String) public type!: string
-  @Prop({ type: String, default: 'blockNumber_DESC' }) public sortBy!: string
-  @Prop({ type: Boolean, default: false }) public listed!: boolean
-  @Prop({ type: Boolean, default: false }) public owned!: boolean
-  @Prop(Boolean) public disableToggle!: boolean
-  @Prop(Boolean) public hideSearch!: boolean
-  @Prop(Boolean) public isMoonRiver!: boolean
-  @Prop(Boolean) public showOwnerSwitch!: boolean
-  @Prop(Array) public sortOption?: string[]
-  protected isVisible = false
-  private preferencesStore = usePreferencesStore()
-
-  public mounted(): void {
-    exist(this.$route.query.search, this.updateSearch)
-    exist(this.$route.query.type, this.updateType)
-    exist(this.$route.query.sort, this.updateSortBy)
-    exist(this.$route.query.listed, this.updateListed)
-    exist(this.$route.query.owned, this.updateOwned)
-  }
-
-  public created() {
-    this.initKeyboardEventHandler({
-      f: this.bindFilterEvents,
-    })
-  }
-
-  private bindFilterEvents(event) {
-    switch (event.key) {
-      case 'b':
-        this.updateListed(!this.vListed)
-        break
-      case 'n':
-        this.updateSortBy(this.sortOption?.[0] || '')
-        break
-      case 'o':
-        this.updateSortBy(this.sortOption?.[1] || '')
-        break
-      case 'a':
-        this.updateSortBy(this.sortOption?.[2] || '')
-        break
-      case 'e':
-        this.updateSortBy(this.sortOption?.[3] || '')
-        break
-    }
-  }
 
-  get vListed(): boolean {
-    return this.listed
-  }
-
-  set vListed(listed: boolean) {
-    this.updateListed(listed)
-  }
-
-  get vOwned(): boolean {
-    return this.owned
-  }
+const emit = defineEmits([
+  'update:listed',
+  'update:owned',
+  'update:type',
+  'update:sortBy',
+  'update:search',
+])
+
+const preferencesStore = usePreferencesStore()
+
+const route = useRoute()
+const router = useRouter()
+const { $consola } = useNuxtApp()
+
+onMounted(() => {
+  exist(route.query.search, updateSearch)
+  exist(route.query.type, updateType)
+  exist(route.query.sortBy, updateSortBy)
+  exist(route.query.listed, updateListed)
+  exist(route.query.owned, updateOwned)
+
+  useKeyboardEvents({
+    f: bindFilterEvents,
+  })
+})
 
-  set vOwned(owned: boolean) {
-    this.updateOwned(owned)
+const bindFilterEvents = (event: KeyboardEvent) => {
+  switch (event.key) {
+    case 'b':
+      updateListed(!vListed.value)
+      break
+    case 'n':
+      updateSortBy(props.sortOption?.[0] || '')
+      break
+    case 'o':
+      updateSortBy(props.sortOption?.[1] || '')
+      break
+    case 'a':
+      updateSortBy(props.sortOption?.[2] || '')
+      break
+    case 'e':
+      updateSortBy(props.sortOption?.[3] || '')
+      break
   }
+}
 
-  get searchQuery(): string {
-    return this.search
-  }
+const vListed = computed({
```

Comment:
`useVmodel`?

Reply:
Hi @roiLeo, i believe we can't use useVModel here as we are not just emitting an `update:listed` here. We are also doing a replaceUrl before emitting. 

But i might be mistaken. Please let me know how you'd want me to do this.